Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,304 @@
+    :map(map), dir(dir), d(d){}
+  
+  template<typename VertexDescriptor, typename U>
+  void operator()(const VertexDescriptor vd, const U&)
+  {
+    typename boost::property_traits<PMAP>::value_type p = get(map, vd) + d*dir;
+    put(map, vd, p);
+  }
+  
+  PMAP map;
+  Vector dir;
+  double d;
+};
+
+struct IdentityFunctor
+{
+  template<typename T, typename U>
+  void operator()(const T&, const U&){}
+};
+}//end internal
+
+/**
+ * \ingroup PMP_meshing_grp
+ * Extrudes `input` into `output` using `bot` and `top`. It means that 
+ * `input` will be copied twice and transformed once using `bot`, once using `top`,
+ * and these two copies will be joined with triangle strips.
+ * @tparam InputMesh a model of the concept `FaceListGraph`
+ * @tparam OutputMesh a model of the concept `FaceListGraph`
+ * @tparam NamedParameters1 a sequence of \ref pmp_namedparameters "Named Parameters" for `InputMesh`
+ * @tparam NamedParameters2 a sequence of \ref pmp_namedparameters "Named Parameters" for `OutputMesh`
+ * @tparam BottomFunctor a functor that will apply a transformation to all points of 
+ * `input` in order to create the first offsetted part of the extrusion. It must have a function 
+ * `void operator()(boost::graph_traits<InputMesh>::vertex_descriptor,
+ * boost::graph_traits<OutputMesh>::vertex_descriptor);
+ * @tparam TopFunctor a functor that will apply a transformation to all points of 
+ * `input` in order to create the second offsetted part of the extrusion. It must have a function 
+ * `void operator()(boost::graph_traits<InputMesh>::vertex_descriptor,
+ * boost::graph_traits<OutputMesh>::vertex_descriptor);
+ * @param input the open triangulated `InputMesh` to extrude.
+ * @param output the `OutputMesh` containing the result of the extrusion.
+ * @param np1 an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map}
+ *    the property map that contains the points associated to the vertices of `input`. 
+ * If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` 
+ * should be available for the vertices of `input` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ * 
+ * * @param np2 an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map}
+ *    the property map that will contain the points associated to the vertices of `output`. 
+ * If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` 
+ * should be available for the vertices of `output` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template <class InputMesh,
+          class OutputMesh,
+          class BottomFunctor,
+          class TopFunctor = internal::IdentityFunctor,
+          class NamedParameters1,
+          class NamedParameters2
+          >
+void generic_extrude_mesh(const InputMesh& input, 
```

Comment:
@CGAL/geometryfactory Can we come up with a better name? It's more a generalization of the extrusion than a generic extrusion.